File: src/main/java/bad/robot/radiate/monitor/MonitoringThreadFactory.java
package bad.robot.radiate.monitor;

import java.util.concurrent.ThreadFactory;
import java.util.concurrent.atomic.AtomicInteger;

public class MonitoringThreadFactory implements ThreadFactory {

    private final AtomicInteger threadCount = new AtomicInteger();

    @Override
    public Thread newThread(Runnable runnable) {
        return new Thread(runnable, "monitoring-thread-" + threadCount.getAndIncrement());
    }
}

File: src/main/scala/bad/robot/radiate/monitor/MonitoringThreadFactory.scala
package bad.robot.radiate.monitor

import java.util.concurrent.ThreadFactory
import java.util.concurrent.atomic.AtomicInteger

class MonitoringThreadFactory extends ThreadFactory {
  private val threadCount = new AtomicInteger

  def newThread(runnable: Runnable): Thread = new Thread(runnable, "radiate-monitoring-thread-" + threadCount.getAndIncrement)
}